app/database.js

```diff
@@ -9,72 +9,206 @@
 		getUserIdbyEmail : "SELECT user_id FROM tblusers WHERE user_email=?",
     getLanguages: "SELECT * FROM tbllanguages",
-		getModules: "SELECT language_id, language_name, subject_id, subject_name FROM tbldatasets,tbllanguages,tblsubjects WHERE dataset_language=language_id AND dataset_subject=subject_id"
+		getModules: "SELECT language_id, language_name, subject_id, subject_name FROM tbldatasets,tbllanguages,tblsubjects WHERE dataset_language=language_id AND dataset_subject=subject_id AND dataset_user=?",
+		replaceUser: "REPLACE INTO tblusers VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)",
+		lastInsertedId: "SELECT ? FROM ? ORDER BY ? DESC LIMIT 1",
+		getSubjectByName : "SELECT * FROM tblsubjects WHERE subject_name=?"
 	};
 
+	var lastId = 0;
+
 	// Check if SQL.js has been loaded through AMD
-	var sql;
+	var sql,db,db_online;
 	if (typeof sqlite !== 'object') {
 		document.body.style.backgroundColor = 'red';
-		alert(lang("error_requirefail", "sql.js"));
+		alert("Failed to require sqlite through AMD.");
 	} else {
 		sql = sqlite;
 	}
 
-	// Initiate DB and check if there is an existing user DB
-	var read_database;
-	if (database_exists(config.constant("DATABASE_USER"))) {
-		read_database = fs.readFileSync(config.constant("DATABASE_USER"));
-	} else {
-		read_database = fs.readFileSync(config.constant("DATABASE_SLIMSTAMPEN"));
-	}
-	var db = new sql.Database(read_database);
-
 	function database_exists(path) {
 		try {
 			fs.accessSync(path, fs.F_OK);
 			return true;
 		} catch (e) {
+			console.log(e);
 			return false;
 		}
 	}
 
-	// Function for Handeling query Error
-	function onError(db, error) {
-		console.log("this error " + error.message);
-		alert('Something went wrong while excuting your request\n please try again! ');
+	function onError(error) {
+		if (error) {
+			if (error.message !== undefined) {
+				console.log(error.message);
+			} else {
+				console.log(error);
+			}
+		}
+	}
+	function isUnique(unique_name1, unique_name2, queryResult, row) {
+	    for (i = 0; i<queryResult.length;i++) {
+	      if (queryResult[i][unique_name1]==row[unique_name1] && queryResult[i][unique_name2]==row[unique_name2]) {
+					return false;
+	      }
+	    }
+	    return true;
 	}
 	
-	// Auxiluary uniqueness function
-  function isUnique(unique_name1, unique_name2, queryResult, row) {
-    for (i = 0; i<queryResult.length;i++) {
-      if (queryResult[i][unique_name1]==row[unique_name1] && queryResult[i][unique_name2]==row[unique_name2]) {
-        return false;
-      }
-    }
-    return true;
-  }
+	function synchronizeUser(userId, callback) {
+		var local_user = database.getQuery('getUser',[userId]);
+		database.getOnlineQuery("getUser", [userId], function(online_user) {
+			online_user = online_user[0];
+			if (local_user.length === 0) {
+				online_user = $.map(online_user, function(val, key) { return (key=="user_createdate" || key=="user_lastedited" || key=="user_bday") ? date.formatDatetime(val) : val; });
```

Comment:
I have no clue as to what is happening here, try to replace it with a descriptive function name.